### PR TITLE
chore: fix inspector item expansion

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

Reported in https://posthog.slack.com/archives/C03PB072FMJ/p1706781876783959

The expanded items in the inspector list were not being cleared in all cases. We cannot maintain id references because we don't have ids :( So the next best thing is to collapse when any change is made

## Changes

Clear expansion settings when any change is made to the inspector search / filtering, and does it consistently now!

https://github.com/PostHog/posthog/assets/6685876/63044dd8-bd3b-4f8a-adf0-a156068388ae

## How did you test this code?

Manually

### CODE REVIEW

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:1]: i wonder if its nicer (and fast enough) to hash the events and use that as an id... then we could search for values _inside_ the events while maintaining the same item open - so e.g. I'm looking at a pageview and search for lib_version it'd be cool to highlight/scroll to lib version in the open event

I may be imagining nice things though

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts [hunk 1]
```diff
@@ -755,6 +759,11 @@ export const playerInspectorLogic = kea<playerInspectorLogicType>([
             }
         },
     })),
+    // subscriptions(({ actions }) => ({
+    //     items: () => {
+    //         actions.setItemExpanded
+    //     },
+    // })),
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:766]:
```suggestion
```

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts [hunk 2]
```diff
@@ -2,6 +2,7 @@ import { customEvent, eventWithTime } from '@rrweb/types'
 import FuseClass from 'fuse.js'
 import { actions, connect, events, kea, key, listeners, path, props, propsChanged, reducers, selectors } from 'kea'
 import { loaders } from 'kea-loaders'
+// import { subscriptions } from 'kea-subscriptions'
```
Comment (pauldambra) [inline frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts:5]:
```suggestion
```

**Review by pauldambra — APPROVED** [review]
does it what it says on the tin...

File: frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts (hunk 2 above)
Comment (daibhin) [reply frontend/src/scenes/session-recordings/player/inspector/playerInspectorLogic.ts]: Thanks, I'd forgotten to push my last commit

**Pull request merged**